packages/core/test/unit/initializer/npm-registry.spec.ts

```diff
@@ -0,0 +1,44 @@
+import { getRegistry } from '../../../src/initializer/npm-registry.js';
+import { expect } from 'chai';
+import log4js from 'log4js';
+
+const DEFAULT_REGISTRY = 'https://registry.npmjs.com';
+
+describe('npm registry', () => {
+  describe('get registry', () => {
+    let oldNpmConfigRegistry: string | undefined;
+    let oldNpmCommand: string | undefined;
+
+    before(() => {
+      oldNpmConfigRegistry = process.env.npm_config_registry;
+
+      oldNpmCommand = process.env.npm_command;
+    });
+
+    after(() => {
+      process.env.npm_config_registry = oldNpmConfigRegistry;
+
+      process.env.npm_command = oldNpmCommand;
+    });
+
+    it('should return default repository when run with npm command with no custom repository', () => {
+      process.env.npm_config_registry = '';
+
+      process.env.npm_command = 'value';
+
+      const registry = getRegistry(log4js.getLogger());
```

Comment:
Please use `testInjector.logger`. It contains a logger mock that is ready to be used. You can verify log messages on it using `sinon.asserts.calledOnceWithExactly(testInjector.logger.warn, '...');`